src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -78,10 +78,10 @@ void goto_symext::symex_allocate(
       if(tmp_type.is_not_nil())
       {
         // Did the size get multiplied?
-        mp_integer elem_size=pointer_offset_size(tmp_type, ns);
+        auto elem_size = pointer_offset_size(tmp_type, ns);
         mp_integer alloc_size;
 
-        if(elem_size<0)
```

Comment:
I think there was some value in the `!= 0` check here as we are going to divide by `elem_size` later on, but that might already be implied by an invariant of `pointer_offset_size()`.